src/main/java/software/amazon/awssdk/crt/http/HttpClientConnection.java

```diff
@@ -23,6 +25,42 @@
  * This class is not thread safe and should not be called from different threads.
  */
 public class HttpClientConnection extends CrtResource {
+    /**
+     * HTTP protocol version.
+     */
+    public enum ProtocolVersion {
+        HTTP_1_0(1),
+        HTTP_1_1(2),
+        HTTP_2(3);
+
+        private int protocolVersion;
+        private static Map<Integer, ProtocolVersion> enumMapping = buildEnumMapping();
+
+        ProtocolVersion(int value) {
+            protocolVersion = value;
+        }
+
+        public static ProtocolVersion getEnumValueFromInteger(int value) {
+            ProtocolVersion enumValue = enumMapping.get(value);
+            if (enumValue != null) {
+                return enumValue;
+            }
+
+            throw new RuntimeException("Illegal signature type value in signing configuration");
+        }
+        private static Map<Integer, ProtocolVersion> buildEnumMapping() {
+            Map<Integer, ProtocolVersion> enumMapping = new HashMap<Integer, ProtocolVersion>();
+            enumMapping.put(HTTP_1_0.getValue(), HTTP_1_0);
+            enumMapping.put(HTTP_1_1.getValue(), HTTP_1_1);
+            enumMapping.put(HTTP_2.getValue(), HTTP_2);
+
+            return enumMapping;
+        }
+
+        public int getValue() {
+            return protocolVersion;
+        }
+    }
```

Comment:
Since it's used outside connection, feel like it's better to move it out as a separate file in http/